biz.aQute.bndlib/src/aQute/bnd/osgi/Builder.java

```diff
@@ -1273,6 +1273,10 @@ public boolean addAll(Jar to, Jar sub, Instruction filter, String destination) {
 			if ("META-INF/MANIFEST.MF".equals(name))
 				continue;
 
+			String parts[] = name.split("/");
```

Comment:
This is an expensive operation to get the last segment of the path. Instead do

```java
if (doNotCopy(Strings.getLastSegment(name, '/'))
```

Reply:
changed as suggested